### PR TITLE
Make `fml/...` compatible with `.clang-tidy`.

### PR DESCRIPTION
**Pull request by matanlurey**

### CODE REVIEW

**Review by cyanglaz — APPROVED** [review]
LGTM, please also update the PR description :)

File: fml/memory/weak_ptr.h [hunk 1]
```diff
@@ -204,7 +204,7 @@ class TaskRunnerAffineWeakPtr {
       T* ptr,
       fml::RefPtr<fml::internal::WeakPtrFlag>&& flag,
       DebugTaskRunnerChecker checker)
-      : ptr_(ptr), flag_(std::move(flag)), checker_(checker) {}
+      : ptr_(ptr), flag_(std::move(flag)), checker_(std::move(checker)) {}
```
Comment (cbracken) [inline fml/memory/weak_ptr.h:207]: Why move it here but pass by const ref above?

Comment (cbracken) [reply fml/memory/weak_ptr.h:207]: Ah never mind; this is a DebugTaskRunnerChecker and the other is a DebugThreadRunner. I had assumed they were the same type. Looking at the top of this file, they look somewhat analogous, differing only in whether they assert we're on the creation thread or creation task runner. Curious why the difference in how they're passed?

Should these types be move-only? I understand what these are trying to accomplish but I'm not familiar with the code in question or its history -- for some reason I had thought we were just using a threadlocal for this.

File: fml/platform/darwin/weak_nsobject.h [hunk 2]
```diff
@@ -217,8 +222,9 @@ class WeakNSObject<id> : public WeakNSProtocol<id> {
  private:
   friend class WeakNSObjectFactory<id>;
 
-  explicit WeakNSObject(RefPtr<fml::WeakContainer> container, debug::DebugThreadChecker checker)
-      : WeakNSProtocol<id>(container, checker) {}
+  explicit WeakNSObject(RefPtr<fml::WeakContainer> container,
+                        const debug::DebugThreadChecker& checker)
+      : WeakNSProtocol<id>(std::move(container), checker) {}
```
Comment (cyanglaz) [inline fml/platform/darwin/weak_nsobject.h:227]: Actually, I didnt catch this first time going through. We should not move here as the factory might spawn multiple weak pointers. Same above for the weak_ptr.

File: fml/memory/weak_ptr.h (hunk 1 above)
Comment (cyanglaz) [reply fml/memory/weak_ptr.h:207]: They should not be moved, the factory should always own a checker to spawn multiple weakPtrs

**Review by cyanglaz — CHANGES_REQUESTED** [review]

File: fml/memory/weak_ptr.h (hunk 1 above)
Comment (matanlurey) [reply fml/memory/weak_ptr.h]: Changed it to a const reference, there isn't a good reason for this to be `std::move`d.

File: fml/platform/darwin/weak_nsobject.h (hunk 2 above)
Comment (matanlurey) [reply fml/platform/darwin/weak_nsobject.h]: Agreed, changing to reference.

**Review by cyanglaz — APPROVED** [review]
LGTM

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

**Comment by cyanglaz** [comment]
I just landed https://github.com/flutter/engine/commit/1427b9c096349572c86b1f0c01c868736fd2c3b8, which is related. Just to be sure, could you rebase to get those changes and re-run the CI?

**Comment by matanlurey** [comment]
> I just landed [1427b9c](https://github.com/flutter/engine/commit/1427b9c096349572c86b1f0c01c868736fd2c3b8), which is related. Just to be sure, could you rebase to get those changes and re-run the CI?

Sure, done.

**Pull request merged**